ipa/ipa_ec2.py

```diff
@@ -228,20 +230,30 @@ def _get_subnet(self, subnet_id):
 
         return subnet
 
+    def _get_user_data(self):
+        """
+        Return cloud init config string.
+
+        The public ssh key is added by cloud init to the instance based on
+        the ssh user and private key file.
+        """
+        key = ipa_utils.generate_public_ssh_key(self.ssh_private_key).decode()
+        data = BASH_SSH_SCRIPT.format(user=self.ssh_user, key=key)
+        return data
```

Comment:
This is not data, the value returned is the actual script, i.e. name it "script"